lib/ansible/plugins/loader.py

```diff
@@ -819,12 +822,13 @@ def _load_module_source(self, name, path):
 
         return module
 
-    def _update_object(self, obj, name, path, redirected_names=None):
+    def _update_object(self, obj, name, path, redirected_names=None, resolved=None):
 
         # set extra info on the module, in case we want it later
         setattr(obj, '_original_path', path)
         setattr(obj, '_load_name', name)
         setattr(obj, '_redirected_names', redirected_names or [])
+        setattr(obj, '_fqcn', resolved)
```

Comment:
tempted to just set the 'name' and aliases at this point

Reply:
yeah, that seems like a better idea

Reply:
need to x2 check, but probably shoudl happen when 'class_only' also (probably except filters/tests)